raysort/sort_utils.py

```diff
@@ -487,3 +487,35 @@ def get_boundaries_auto(
         cfg.num_reducers_per_worker,
         get_boundaries_impl=get_boundaries_impl,
     )
+
+
+def get_median_key(part: np.ndarray) -> float:
+    records = part.reshape((-1, 100))
+    key_bytes = records[:, :8]
+    keys = key_bytes.view(dtype=np.dtype('>u8'))
+    return np.median(keys)
+
+@ray.remote(num_returns=2)
+def split_part(part, pivot, identifier):
+    def id_print(*output):
+        print(":)) split_part: identifier:", identifier, "| output:", output)
+
+
+    if isinstance(part, ray.ObjectRef):        
```

Comment:
`part` shouldn't be an ObjectRef. Can you confirm and remove this check?

Reply:
I added it because we don't call ray.get on all the parts in final_merge, only the first one, so the other parts will be objectrefs when passed in.